tests/unit/test_projection_utilities.py

```diff
@@ -1228,11 +1240,90 @@ def test_get_x_y_extents_from_geographic_perimeter_full_earth_laea(self):
 
         points1 = [(-180, -90), (-180, 90), (180, 90), (180, -90)]
         x_y_extents = get_x_y_extents_from_geographic_perimeter(
-            points1, crs, x_values, y_values
+            points1, crs, granule_extent_projected_meters
         )
 
         self.assertTrue(all(not math.isinf(value) for value in x_y_extents.values()))
 
+    def test_remove_any_invalid_projected_values(self):
+        """Ensure that only valid values in the x,y list are returned and any
+        NaN or inf values are removed.
+        """
+        points_x = [
+            float('-inf'),
+            -900.3,
+            -800.2,
+            -700.1,
+            float('nan'),
+            500.0,
+            600.9,
+            700.0,
+            800.0,
+        ]
+        points_y = [
+            89.0,
+            float('nan'),
+            69.1,
+            40.5,
+            -10.6,
+            50.9,
+            70.2,
+            80.4,
+            float('inf'),
+        ]
+        expected_x_values = np.array([-800.2, -700.1, 500.0, 600.9, 700.0])
+        expected_y_values = np.array([69.1, 40.5, 50.9, 70.2, 80.4])
+        valid_x, valid_y = remove_any_invalid_projected_values(points_x, points_y)
+        np.array_equal(valid_x, expected_x_values)
+        np.array_equal(valid_y, expected_y_values)
+
+    def test_check_perimeter_exceeds_grid_extents(self):
+        """Ensure that True value is returned when the input perimeter array exceeds grid
+        extents and a False value is returned when the perimeter is within the grid extents
+
+        """
+        granule_extent = {"x_min": -1000, "x_max": 1000, "y_min": -1000, "y_max": 1000}
+        self.assertTrue(
+            check_perimeter_exceeds_grid_extents(
+                np.array([-2000.1, -1000.1, -900, -500, 200.3, 700.1, 800.1, 1200.5]),
+                np.array([1100.1, 400.9, 200, -100.8, -300.3, -500.1, -600.1, -1000.5]),
+                granule_extent,
+            )
+        )
+
+        self.assertFalse(
+            check_perimeter_exceeds_grid_extents(
+                np.array([-2000.1, -1000.1, -900, -500, 200.3, 700.1, 800.1, 900.0]),
+                np.array([400.1, 300.9, 200, -100.8, -300.3, -500.1, -600.1, -700.5]),
+                granule_extent,
```

Comment:
This confused me because I thought this should fail.  But it's because I didn't fully understand the function. Can you rename to `perimeter_surrounds_grid`? 
Would that make more sense?  If not, (or in any case) can you include that in the function comment 

```
Check if perimeter sourrounds the grid completely, e.g. that perimeter exceeds the grid extents on all axes.
```